dotenv/src/errors.rs

```diff
@@ -1,16 +1,37 @@
-use std::io;
-use std::fmt;
 use std::error;
+use std::fmt;
+use std::io;
 
-pub type Result<T> = std::result::Result<T, Error>;
+pub type Result<T = ()> = std::result::Result<T, Error>;
+
+#[derive(Debug)]
+pub struct ParseError(pub String, pub usize);
+
+impl ParseError {
+    pub fn new(line: String, index: usize) -> ParseError {
+        ParseError(line, index)
+    }
+}
+
+impl fmt::Display for ParseError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            fmt,
+            "Error parsing line: '{}', error at line index: {}",
+            self.0, self.1
+        )
+    }
+}
+
+impl error::Error for ParseError {}
 
 #[derive(Debug)]
 pub enum Error {
-    LineParse(String, usize),
+    LineParse(ParseError),
```

Comment:
What is the logic behind introducing a new layer of indirection here?

Reply:
i think this enhance readability and Integrity code what you think ?